### PR TITLE
crd: add certificate authority

### PR DESCRIPTION
**Pull request by calebdoxsey**
## Summary
Add `caSecret` to the CRD.

## Related issues
Fixes https://github.com/pomerium/ingress-controller/issues/476

## Checklist
- [x] reference any related issues
- [ ] updated unit tests
- [x] add appropriate tag (`improvement` / `bug` / etc)
- [x] ready for review

### CODE REVIEW

**Comment by wasaga** [comment]
could we make it to accept an array of CA secrets and glue them, as was mentioned in https://github.com/pomerium/ingress-controller/issues/476#issuecomment-1375901922 

we currently implicitly support multiple CAs as PEM allows for multiple certs as they're added via https://pkg.go.dev/crypto/x509#CertPool.AppendCertsFromPEM that supports multiple certs: https://github.com/pomerium/pomerium/blob/04a82813f3e884bc1fe161ad87ec60ffa2cc9e9e/pkg/cryptutil/tls.go#L40

in k8s environment, each CA would be located in a dedicated secret and supporting multiple would require user to somehow monitor them for updates and manually glue them into yet another secret.

File: pomerium/ctrl/bootstrap.go [hunk 1]
```diff
@@ -26,6 +26,7 @@ func Apply(ctx context.Context, dst *config.Options, src *model.Config) error {
 		fn   func(context.Context, *config.Options, *model.Config) error
 	}{
 		{"secrets", applySecrets},
+		{"ca", applyCertificateAuthority},
```
Comment (wasaga) [inline pomerium/ctrl/bootstrap.go:29]: bootstrap config surface was intentionally kept to bare minimum - is there a special reason CA should be part of it and not set via the databroker config like other options?

Comment (calebdoxsey) [reply pomerium/ctrl/bootstrap.go:29]: sorry I don't understand the distinction here. I thought this was where I needed to set options. I don't think I understand how to make this change.

Comment (wasaga) [reply pomerium/ctrl/bootstrap.go:29]: There are two reconcilers running watching for changes in the CRD. 

One is controlling core pomerium (the `pomerium/ctrl`) and makes a bare minimum config to keep it up and running and accepting gRPC databroker parameters. 

The other one sets full set of parameters via the databroker config - see https://github.com/pomerium/ingress-controller/blob/main/pomerium/config.go - my suggestion is to move handling of this CA there.

**Pull request closed without merging**